src/zenml/model_registries/base_model_registry.py

```diff
@@ -0,0 +1,425 @@
+            version: The version of the model version.
+            version_description: The description of the model version.
+            version_tags: The tags associated with the model version.
+            registry_metadata: The metadata associated with the model
+                version.
+            zenml_version: The ZenML version of the model version.
+            zenml_pipeline_run_id: The ZenML pipeline run ID of the model
+                version.
+            zenml_pipeline_name: The ZenML pipeline run name of the model
+                version.
+            zenml_step_name: The ZenML step name of the model version.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            The registered model version.
+        """
+
+    @abstractmethod
+    def delete_model_version(
+        self,
+        name: str,
+        version: str,
+    ) -> None:
+        """Deletes a model version from the model registry.
+
+        Args:
+            name: The name of the registered model.
+            version: The version of the model version to delete.
+        """
+
+    @abstractmethod
+    def update_model_version(
+        self,
+        name: str,
+        version: str,
+        version_description: Optional[str] = None,
+        version_tags: Optional[Dict[str, str]] = None,
+        version_stage: Optional[ModelVersionStage] = None,
+    ) -> ModelVersion:
+        """Updates a model version in the model registry.
+
+        Args:
+            name: The name of the registered model.
+            version: The version of the model version to update.
+            version_description: The description of the model version.
+            version_tags: The tags associated with the model version.
+            version_stage: The stage of the model version.
+
+        Returns:
+            The updated model version.
+        """
+
+    @abstractmethod
+    def list_model_versions(
+        self,
+        name: Optional[str] = None,
+        model_source_uri: Optional[str] = None,
+        version_tags: Optional[Dict[str, str]] = None,
+        **kwargs: Any,
+    ) -> List[ModelVersion]:
+        """Lists all model versions for a registered model.
+
+        Args:
+            name: The name of the registered model.
+            model_source_uri: The model source URI of the registered model.
+            version_tags: The tags associated with the registered model.
+            kwargs: Additional keyword arguments.
+
+        Returns:
+            A list of model versions.
+        """
+
+    @abstractmethod
+    def get_model_version(self, name: str, version: str) -> ModelVersion:
+        """Gets a model version for a registered model.
+
+        Args:
+            name: The name of the registered model.
+            version: The version of the model version to get.
+
+        Returns:
+            The model version.
+        """
+
+    @abstractmethod
+    def check_model_version_exists(
```

Comment:
Same thing here: this should be handled by raising `KeyError` in `get_model_version`.